issues/order.ts

```diff
@@ -94,8 +99,7 @@ export class OrderQueue {
         if (!order.cancelled && order.date > expirationDate) {
           await this.processor.orderProcessor(order);
 
-          const emails = order.recipients.split(';');
-          emails.forEach(async (email) => {
+          order.recipients.forEach(async (email) => {
```

Comment:
There's a catch with Promise handling here that is not easy to spot.

When using `forEach` you're not waiting for the promises to be fulfilled. This means that the next order is processed before the mails are delivered.

One solution would be to use something like this:

```
await Promise.all(
  order.recipients.forEach(async (email) => {
    await this.processor.sendConfirmationMail(order, email);
  }
);
```

Here you're mapping all recipients into promises and then waiting for the Promises to be resolved.

Even with these changes it would still be missing proper error handling.